core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/convert.kt

```diff
@@ -76,41 +77,49 @@ internal fun AnyCol.convertToTypeImpl(to: KType): AnyCol {
             nullsFound = true
             null
         }
+
         else -> throw TypeConversionException(null, from, to)
     }
 
-    return when {
-        from == to -> this
-        from.isSubtypeOf(to) -> (this as DataColumnInternal<*>).changeType(to.withNullability(hasNulls()))
-        else -> when (val converter = getConverter(from, to, ParserOptions(locale = Locale.getDefault()))) {
-            null -> when (from.classifier) {
-                Any::class, Number::class, java.io.Serializable::class -> {
-                    // find converter for every value
-                    val values = values.map {
-                        it?.let {
-                            val clazz = it.javaClass.kotlin
-                            val type = clazz.createStarProjectedType(false)
-                            val converter = getConverter(type, to, ParserOptions(locale = Locale.getDefault())) ?: throw TypeConverterNotFoundException(from, to)
-                            converter(it)
-                        }.checkNulls()
-                    }
-                    DataColumn.createValueColumn(name, values, to.withNullability(nullsFound))
-                }
-                else -> throw TypeConverterNotFoundException(from, to)
-            }
-            else -> {
+    if (from == to) return this
+
+    if (from.isSubtypeOf(to)) try {
+        return (this as DataColumnInternal<*>).changeType(to.withNullability(hasNulls()))
+    } catch (e: UnsupportedOperationException) { /* */
```

Comment:
do you remember why?

Reply:
Yes, I believe this fixed a bug where converting from ColumnGroupImpl or FrameColumnImpl to something would crash because it doesn't have the `changeType` option. However, `convertTo` can now support conversions for any type of column, so the exception is caught here to give any user-given conversions a try first, otherwise, a TypeConverterNotFoundException is thrown.